Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift

```diff
@@ -5,331 +5,61 @@
-            ]
-        }
-
-        // Giphy
-        if options.contains(.giphy) {
-            let giphyView = createGiphyView()
-            contentContainer.addArrangedSubview(giphyView, respectsLayoutMargins: false)
-            constraintsToActivate += [
-                // This is ugly. Ideally the photo preview should be updated to fill all available space.
-                giphyView.widthAnchor.constraint(equalToConstant: window!.bounds.width * 0.75).almostRequired
-            ]
-        }
-
-        // Text
-        if options.contains(.text) {
-            let textView = createTextView()
-            contentContainer.addArrangedSubview(textView, respectsLayoutMargins: true)
-        }
-
-        // File previews
-        if options.contains(.filePreview) {
-            let filePreviewView = createFilePreviewView()
-            contentContainer.addArrangedSubview(filePreviewView, respectsLayoutMargins: false)
-        }
-
-        // Link preview
-        if options.contains(.linkPreview) {
-            let linkPreviewView = createLinkPreviewView()
-            contentContainer.addArrangedSubview(linkPreviewView, respectsLayoutMargins: true)
-            constraintsToActivate += [
-                // This is ugly. Ideally the link preview should be updated to fill all available space.
-                linkPreviewView.widthAnchor.constraint(equalToConstant: window!.bounds.width * 0.75).almostRequired
-            ]
-        }
-
-        // Actions
-        if options.contains(.actions) {
-            let actionsView = createActionsView()
-            contentContainer.addArrangedSubview(actionsView, respectsLayoutMargins: false)
-            constraintsToActivate += [
-                // This is ugly. Ideally the action view should be updated to fill all available space.
-                actionsView.widthAnchor.constraint(equalToConstant: window!.bounds.width * 0.75).almostRequired
-            ]
-        }
-
-        // Reactions
-        if options.contains(.reactions) {
-            let reactionsView = createReactionsView()
-            reactionsView.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
-            reactionsView.setContentCompressionResistancePriority(.defaultHigh, for: .vertical)
-            contentView.addSubview(reactionsView)
-
-            constraintsToActivate += [
-                reactionsView.topAnchor.pin(equalTo: contentView.topAnchor),
-                reactionsView.centerXAnchor.pin(
-                    equalTo: options.contains(.flipped) ? bubbleView.leadingAnchor : bubbleView.trailingAnchor
-                ),
-                bubbleView.topAnchor.pin(equalTo: reactionsView.centerYAnchor)
-            ]
-        }
-        
-        NSLayoutConstraint.activate(constraintsToActivate)
+    var content: _ChatMessage<ExtraData>? {
+        get { messageContentView.content }
+        set { messageContentView.content = newValue }
     }
```

Comment:
Let's not do this. These values should be accessed directly on the view

Reply:
fixed 👍